encord/orm/dataset.py

```diff
@@ -117,28 +276,212 @@ def created_at(self) -> datetime:
+        return self["file_link"]
+
+    @property
+    def signed_url(self) -> Optional[str]:
+        """
+        The cached signed url of the given data asset. To cache the signed url, use the
+        :meth:`~encord.orm.dataset.DataRow.fetch()` function.
+        """
+        return self["signed_url"]
+
+    @property
+    def file_size(self) -> int:
+        """
+        The file size of the given data asset in bytes.
+        """
+        return self["file_size"]
+
+    @property
+    def file_type(self) -> str:
+        """
+        A MIME file type of the given data asset as a string
+        """
+        return self["file_type"]
+
+    @property
+    def storage_location(self) -> StorageLocation:
+        return self["storage_location"]
+
+    @property
+    def images(self) -> Optional[List[ImageData]]:
+        """
+        A list of the cached :class:`~encord.orm.dataset.ImageData` objects for the given data asset.
+        Fetch the images with appropriate settings in the :meth:`~encord.orm.dataset.DataRow.fetch()` function.
+        If the data type is not :meth:`DataType.IMG_GROUP <encord.constants.enums.DataType.IMG_GROUP>`
+        then this returns None.
+        """
+        return self["images"]
+
+    @property
+    def dicom_signed_urls(self) -> Optional[List[str]]:
+        """
+        The signed urls of all the DICOM files in the series of the :class:`~encord.orm.dataset.DataRow`.
+        Returns `None` if the data type is not :meth:`DataType.DICOM <encord.constants.enums.DataType.DICOM>`.
+        """
+        return self["dicom_signed_urls"]
+
+    @property
+    def is_optimised_image_group(self) -> Optional[bool]:
+        """
+        If the data type is an :meth:`DataType.IMG_GROUP <encord.constants.enums.DataType.IMG_GROUP>`,
+        returns whether this is a performance optimised image group. Returns `None` for other data types.
+        """
+        return self["is_optimised_image_group"]
+
+    def fetch(
+        self,
+        *,
+        signed_url: bool = False,
+        images: Optional[FetchImagesDataConfig] = None,
+        client_metadata: bool = False,
+    ):
+        """
+        Fetches all the most up-to-date data. If any of the parameters are falsy, the current values will not be
+        updated.
+
+        Args:
+            signed_url: If True, this will fetch a generated signed url of the data asset.
+            images: If not None, this will fetch the image data of the data asset. You can additionally
+                specify what to fetch with the :class:`.FetchImagesDataConfig` class.
+            client_metadata: If True, this will fetch the client metadata of the data asset.
+        """
+        if self["_querier"] is not None:
+            fetch_images = None
+            if images is not None:
+                fetch_images = dataclasses.asdict(images)
+
+            payload = {
+                "fetch_signed_url": signed_url,
+                "fetch_images": fetch_images,
```

Comment:
[curious]🤔

What would happen if the data row is the image and we pass these options?

Reply:
So the BE is set out to just ignore "wrong" arguments. I could also throw in such cases which would be more explicit for sure, maybe a bit more annoying. What would you expect?

Reply:
I think explicitness would be preferable in this case, just to prevent a misunderstanding from the user's perspective. ("I have used this config but it doesn't do anything")

Reply:
I talked to @alexey-cord-tech about this (IIRC) and he was of the opinion that if we iterate over a dataset with different data types, it is better that these requests would not intermittently throw, but just ignore the "wrong" values. I'll leave it in that case.